internal/apis/certmanager/v1alpha2/types_certificate.go

```diff
@@ -332,8 +332,27 @@ type PKCS12Keystore struct {
 	// PasswordSecretRef is a reference to a key in a Secret resource
 	// containing the password used to encrypt the PKCS12 keystore.
 	PasswordSecretRef cmmeta.SecretKeySelector `json:"passwordSecretRef"`
+
+	// Algorithm is the encryption and MAC algorithms used to create the PKCS12 keystore.
+	//
+	// If provided, allowed values are either `RC2-40-CBC:HMAC-SHA-1` or `AES-256-CBC:HMAC-SHA-2`.
+	// Default value is `RC2-40-CBC:HMAC-SHA-1` for backward compatibility.
+	// Note: By default, OpenSSL 3 can't decode PKCS#12 files created using `RC2-40-CBC:HMAC-SHA-1`.
+	// +optional
+	Algorithm PKCS12Algorithm `json:"algorithm,omitempty"`
 }
 
+// +kubebuilder:validation:Enum="RC2-40-CBC:HMAC-SHA-1";"AES-256-CBC:HMAC-SHA-2"
+type PKCS12Algorithm string
+
+const (
+	// PBE with RC2 certificate algorithm, PBE with 3DES key algorithm and HMAC-SHA-1 MAC algorithm.
+	RC2PKCS12Algorithm PKCS12Algorithm = "RC2-40-CBC:HMAC-SHA-1"
```

Comment:
Maybe a dumb question, but why do we need a MAC algorithm, is the keystore not just encrypted?
Where can I find more about where this MAC is used to create this keystore?

Reply:
The definition of the encoder are given in `go-pkcs12`:
- RC2: https://github.com/SSLMate/go-pkcs12/blob/master/pkcs12.go#L80
- AES256: https://github.com/SSLMate/go-pkcs12/blob/master/pkcs12.go#L139
(I think the MAC provides integrity protection and ensures that the PKCS#12 file has not been tampered after it was created.)

If you want to be more concise, `RC2-40-CBC` (or `RC2`) and `AES-256-CB` (or `AES256`) with corresponding documentation may be more than adequate.

Another question would be whether cert-manager also want to offer the `pkcs12.LegacyDES` encoder as an additional option?

Reply:
I do remember LegacyDES would be a good alternative to RC2 because it is more compatible than eg. AES.

Reply:
Interesting, HMAC-SHA-1 is used both for key derivation and MAC.

Reply:
Using LegacyDES as a new default might compromise backwards compatibility, but I will add it as another option.

Do you have a preference for the naming of the algorithms?

Reply:
I would prefer the shorter names (RC2, DES3 and AES256).
Also we should add a comment that explains when to use what:
- RC2: deprecated, should only be used for backwards compatibility (not supported by openssl 3)
- DES3: insecure, should be used for maximal compatibility
- AES256: should be used when indicated by policy, non-encrypted keystore (PEM format) still lives in the same Secret